src/Model/AutoSegmentation/AutoSegmentation.py

```diff
@@ -90,25 +91,31 @@ def _prepare_output_paths(self) -> tuple[str, str]:
 
     def _copy_temp_dicom_dir(self) -> None:
         """
-        Copies the DICOM directory to a temporary location for processing.
-        This method creates a temporary directory and copies the DICOM files
-        into it, excluding RTSTRUCT files.
+            Copies the file paths from patient dict to a temporary location for processing.
+            This method creates a temporary directory and copies the DICOM files paths
+            into it, excludes rtdose, rtplan, adn rtstruct files.
 
-        Raises:
-            ValueError: If the DICOM directory is not set or copying fails.
-        """
-        if not self.dicom_dir:
-            raise ValueError(f"No dicom directory found: {self.dicom_dir!r}")
+            Raises:
+                ValueError: If the DICOM directory is not set or copying fails.
+         """
+        if not self.file_paths:
+            raise ValueError(f"No dicom files found in {self.file_paths}")
```

Comment:
We should start thinking about creating custom Error classes.  Not necessary now, but worth thinking about.